client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -118,23 +121,27 @@ const Checkout = React.createClass( {
 	},
 
 	addProductToCart: function() {
-		const planSlug = getUpgradePlanSlugFromPath( this.props.product, this.props.selectedSite );
-
-		let cartItem,
-			cartMeta;
-
-		if ( planSlug ) {
-			cartItem = getCartItemForPlan( planSlug );
+		const { product, productsList, purchaseId, selectedSite, selectedSiteSlug } = this.props;
+		const meta = product.split( ':' )[ 1 ];
+		const productSlug = getUpgradePlanSlugFromPath( product, selectedSite.jetpack ) || product.split( ':' )[ 0 ];
+
+		let cartItem = getCartItemForPlan( productSlug );
+
+		if ( 'theme' === productSlug ) {
+			cartItem = themeItem( meta );
+		} else if ( 'domain-mapping' === productSlug ) {
+			cartItem = domainMapping( { domain: meta } );
+		} else if ( 'no-ads' === productSlug ) {
+			cartItem = noAdsItem();
+		} else if ( meta ) {
+			cartItem = Object.assign( { meta }, productsList.data[ productSlug ] );
 		}
 
-		if ( startsWith( this.props.product, 'theme' ) ) {
-			cartMeta = this.props.product.split( ':' )[ 1 ];
-			cartItem = themeItem( cartMeta );
-		}
-
-		if ( startsWith( this.props.product, 'domain-mapping' ) ) {
-			cartMeta = this.props.product.split( ':' )[ 1 ];
-			cartItem = domainMapping( { domain: cartMeta } );
+		if ( cartItem && purchaseId ) {
+			cartItem = cartItems.getRenewalItemFromCartItem( cartItem, {
+				id: purchaseId,
+				domain: selectedSiteSlug
```

Comment:
We're using the slug representation of the domain here. In the slug representation, forward slashes `/` are converted to double colons `::`. Can we verify that were not introducing a change with the current behavior? Maybe add a couple of tests for that?